### DISCUSSION TITLE
How to use Controller ?

### DISCUSSION THREAD

**Discussion by lheido** [pmndrs/react-spring#1690] [Support]
## 🐛 Bug Report

Hi guys, first I'm new to the React world.
I've been trying to use the Controller class without success.
I just copy/paste the example in the [documentation](https://react-spring.io/classes/controller) but the div never fades.
I don't understand what the problem is.

## To Reproduce

Copy/paste the Controller example in the documentation to a sandbox (CodeSandbox).

## Link to repro

https://codesandbox.io/s/cocky-chebyshev-p7lhr?file=/src/Test.js

## Environment

- `react-spring` v9.2.4
- `react` v17.0.2

**Comment by joshuaellis** [comment]
What's your use case for using the Controller class? 😄

**Comment by lheido** [comment]
I plan to use it with the react-use-gesture package. The render-props using `<Spring>`  doesn't seem to be the right way for me.

**Comment by joshuaellis** [comment]
Well you need to pass the `Controller.springs` to the Animated component, then you need to use the Imperative API methods documented [here](https://react-spring.io/common/imperatives-and-refs#api-methods) to use the controller effectively.

I've updated your example to show you it all working https://codesandbox.io/s/epic-thunder-8cpii
